### PR TITLE
Added support for image integral computation.

### PR DESCRIPTION
**Pull request by rold2007**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
This pull request replaces [this one](https://github.com/SixLabors/ImageSharp/pull/1441).

### CODE REVIEW

File: src/ImageSharp/Processing/Extensions/Transforms/ImagingUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Processing
+{
+    /// <summary>
+    /// Defines extensions that allow the computation of image integrals on an <see cref="Image"/>
+    /// </summary>
+    public static class ImagingUtilities
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Processing/Extensions/Transforms/ImagingUtilities.cs:15]: The only thing I would change is to make the existing `ProcessingExtensions` partial and rename this one to match. `ProcessingExtensions.IntegralImage.cs`. I'd also move it to the same `Processing/Extensions/` folder.

File: src/ImageSharp/Processing/Extensions/Transforms/ImagingUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        /// <summary>
+        /// Apply an image integral. See https://en.wikipedia.org/wiki/Summed-area_table
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Processing/Extensions/Transforms/ImagingUtilities.cs:18]: You can use `<see href="https://en.wikipedia.org/wiki/Summed-area_table"/>` to better work with intellisense docs.

File: tests/ImageSharp.Tests/Processing/Transforms/ImagingUtilitiesTests.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+using Xunit;
+
+namespace SixLabors.ImageSharp.Tests.Processing.Transforms
```
Comment (JimBobSquarePants) [inline tests/ImageSharp.Tests/Processing/Transforms/ImagingUtilitiesTests.cs:9]: Move to `tests/ImageSharp.Tests/Processing/` and rename class.

**Review by JimBobSquarePants — CHANGES_REQUESTED** [review]
Looks good. Thanks for resubmitting after I messed up!

Just some minor organizations changes, the code looks good 👍

**Comment by JimBobSquarePants** [comment]
Sorry @rold2007 

We just added Git LFS to the repository and it's rewritten the history. This has forced the PR to be closed and I cannot reopen it. 

If you like I can create a new PR based upon your work to save you the effort of redoing it?

**Comment by rold2007** [comment]
@JimBobSquarePants No worries. I'll open a new PR myself.

**Comment by JimBobSquarePants** [comment]
Thanks @rold2007 !!

**Pull request closed without merging**